pkg/controller/replication_controller.go

```diff
@@ -204,6 +204,10 @@ func (rm *ReplicationManager) getPodControllers(pod *api.Pod) *api.ReplicationCo
 // When a pod is created, enqueue the controller that manages it and update it's expectations.
 func (rm *ReplicationManager) addPod(obj interface{}) {
 	pod := obj.(*api.Pod)
+	if pod.DeletionTimestamp != nil {
```

Comment:
Discussed irl. The point of putting the rc to sleep till the watch has propogated adds/delets is so we get confirmation that the rpcs initiated by the rc has propogated throughout the system before we take further action (we're assuming all other watcers are equally fast here). 

Please add a comment saying something like (probably rephrased): when a pod is deleted gracefully it's deletion timestamp is first modified to reflect a grace period, and after such time has passed, the kubelet actually deletes it from the store. We receive an update for modification of the deletion timestamp and expect an rc to create more replicas asap, not wait until the kubelet actually deletes the pod. This is different from the Phase of a pod changing, because an rc never initiates a phase change, and so is never asleep waiting for the same. 

The phase of a pod changing while an rc is creating replicas is addressed in https://github.com/GoogleCloudPlatform/kubernetes/issues/7369.